compiler/src/main/kotlin/visitors/typechecker.kt

```diff
@@ -408,6 +457,34 @@ class TypeChecker(symbolTable: Table) : ScopedASTVisitor(symbolTable = symbolTab
     override fun visit(node: FuncCallExpr) {
         super.visit(node)
 
+        // Get function declaration for type-checking
+        val funcDecl = (symbolTableSession.getSymbol(node.ident) as FuncDecl)
```

Comment:
The identifier might not be a FuncDecl so this could creating casting exceptions. This should also be checked by the typechecker, and it should raise an error saying something like "ident cannot be called since it is not a function". It would be relevant in this program:
```
// omitted
state Alive (0, 0, 0) {
    let x = 0;
    let y = x();
}
```

Reply:
Fixed in 9314955. Has test exactly as suggested.